File: April-18/25-04-18/python/01-KK-and-Crush.py
import sys

inputData = []

for line in sys.stdin:
    for item in line.split():
        inputData.append(int(item))
            


N = 100001

cnt = 1
for _ in range(0, inputData[0]):
    n = inputData[cnt]
    cnt += 1
    neg = [0]*N
    pos = [0]*N
    
    for i in range(0,n):
        if(inputData[cnt] < 0):
            neg[abs(inputData[cnt])] = 1
        else:
            pos[inputData[cnt]] = 1
        cnt += 1
    
    q = inputData[cnt]
    cnt += 1
    
    for i in range(0, q):
        x = inputData[cnt]
        if(x < 0):
            if(neg[abs(x)] == 1):
                print("Yes")
            else:
                print("No")
        else:
            if(pos[x] == 1):
                print("Yes")
            else:
                print("No")
            
        cnt += 1

File: April-18/25-04-18/c++/01-KK-and-Crush.cpp
#include<bits/stdc++.h>

using namespace std;
const int N = 100001;

int main()
{
    int t;
    scanf("%d", &t);
    while(t--)
    {
        int n;
        scanf("%d", &n);
        int pos[N] = {0},neg[N] = {0};
        for(int i = 0; i < n; i++)
        {
            int a;
            scanf("%d", &a);
            if(a < 0)
                neg[abs(a)] = 1;
            else
                pos[a] = 1;
        }
        int q;
        cin >> q;
        for(int i = 0; i < q; i++)
        {
            int x;
            cin >> x;
            if(x < 0)
            {
                if(neg[abs(x)] == 1)
                    printf("Yes\n");
                else
                    printf("No\n");
            }
            else
            {
                if(pos[x] == 1)
                    printf("Yes\n");
                else
                    printf("No\n");
            }
        }
        
    }
}
